Core.Collectors/IO/AzureHelpers.cs

```diff
@@ -269,5 +275,51 @@ public static string GenerateNotificationMessage(CloudBlob blob)
             string path = blobPath.Split($"{container}/").Last();
             return Tuple.Create(container, path);
         }
+
+        public static CloudStorageAccount GetStorageAccount(string storageConnectionEnvironmentVariable)
+        {
+            string stagingBlobConnectionString = Environment.GetEnvironmentVariable(storageConnectionEnvironmentVariable);
+            return CloudStorageAccount.Parse(stagingBlobConnectionString);
+        }
+
+        public static async Task<CloudStorageAccount> GetStorageAccountUsingMsiAsync(string storageAccountNameEnvironmentVariable, ITelemetryClient telemetryClient, bool isQueue = false)
+        {
+            string storageAccountName = Environment.GetEnvironmentVariable(storageAccountNameEnvironmentVariable);
+            string resource = isQueue ? GetQueueResource(storageAccountName) : GetBlobResource(storageAccountName);
+            StorageCredentials creds = await GetStorageCredentails(resource, telemetryClient).ConfigureAwait(false);
+            CloudStorageAccount storageAccount = new CloudStorageAccount(creds, storageAccountName, EndPointSuffix, true);
+            return storageAccount;
+        }
+
+        private static async Task<StorageCredentials> GetStorageCredentails(string resource, ITelemetryClient telemetryClient)
+        {
+            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            string token = await azureServiceTokenProvider.GetAccessTokenAsync(resource).ConfigureAwait(false);
+            var jwt = new JwtSecurityToken(token);
+            tokenExpiration = jwt.ValidTo;
+            Dictionary<string, string> properties = new Dictionary<string, string>()
+            {
+                { "Resource", resource },
+                { "TokenValidTo", tokenExpiration.ToString()}
+            };
+            telemetryClient.TrackEvent("MsiTokenGeneration", properties);
+            TokenCredential tokenCredential = new TokenCredential(token);
+            return new StorageCredentials(tokenCredential);
+        }
+
+        public static string GetBlobResource(string storageAccountName)
+        {
+            return $"https://{storageAccountName}.blob.core.windows.net/";
+        }
+
+        public static string GetQueueResource(string storageAccountName)
+        {
+            return $"https://{storageAccountName}.queue.core.windows.net/";
+        }
+
+        public static DateTime GetTokenExpiration()
+        {
+            return tokenExpiration;
+        }
```

Comment:
Where is this used? Somewhere else to decide? I would rather have this completely contained within this class, and potentially no variable (remember I don't like state in these classes :))

Reply:
I'm using this static variable instead in ROS. I removed this method altogether